### ISSUE TITLE
Error occurred in cross compilation[Bug] 

### ISSUE THREAD

**Issue by Dark1Forest** [open-mmlab/mmdeploy#1796]
### Checklist

- [X] I have searched related issues but cannot get the expected help.
- [X] 2. I have read the [FAQ documentation](https://github.com/open-mmlab/mmdeploy/blob/master/docs/en/faq.md) but cannot get the expected help.
- [X] 3. The bug has not been fixed in the latest version.

### Describe the bug

An error is reported when the following command is executed
`
sh -x tools/scripts/ubuntu_cross_build_aarch64.sh
`

### Reproduction

sh -x tools/scripts/ubuntu_cross_build_aarch64.sh

### Environment

```Shell
python tools/check_env.py
Traceback (most recent call last):
  File "/mnt/workspace/mmdeploy/tools/check_env.py", line 4, in <module>
    from mmcv.utils import get_git_hash
ImportError: cannot import name 'get_git_hash' from 'mmcv.utils' (/mnt/assembly/anaconda3/envs/torch2/lib/python3.9/site-packages/mmcv/utils/__init__.py)
```

### Error traceback

```Shell
+ make -j29
Consolidate compiler generated dependencies of target mmdeploy_ncnn_ops_obj
Consolidate compiler generated dependencies of target spdlog
[  3%] Built target mmdeploy_ncnn_ops_obj
[  7%] Built target spdlog
[  8%] Built target mmdeploy_ncnn_ops
Consolidate compiler generated dependencies of target mmdeploy_core
[  8%] Building CXX object csrc/mmdeploy/core/CMakeFiles/mmdeploy_core.dir/module.cpp.o
[  9%] Building CXX object csrc/mmdeploy/core/CMakeFiles/mmdeploy_core.dir/model.cpp.o
[ 10%] Building CXX object csrc/mmdeploy/core/CMakeFiles/mmdeploy_core.dir/net.cpp.o
[ 11%] Building CXX object csrc/mmdeploy/core/CMakeFiles/mmdeploy_core.dir/graph.cpp.o
In file included from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/net.h:7,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/net.cpp:3:
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/net.cpp: In function ‘mmdeploy::_registry::Registry<mmdeploy::framework::Net>& mmdeploy::framework::get_registry(mmdeploy::basic_type<mmdeploy::framework::Net>)’:
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/registry.h:211:47: error: use of deleted function ‘mmdeploy::_registry::Registry<void>::Registry(const mmdeploy::_registry::Registry<void>&)’
  211 |     static ::mmdeploy::Registry<tag> instance{};                         \
      |                                               ^
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/net.cpp:9:1: note: in expansion of macro ‘MMDEPLOY_DEFINE_REGISTRY’
    9 | MMDEPLOY_DEFINE_REGISTRY(Net);
      | ^~~~~~~~~~~~~~~~~~~~~~~~
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/registry.h:101:20: note: ‘mmdeploy::_registry::Registry<void>::Registry(const mmdeploy::_registry::Registry<void>&)’ is implicitly deleted because the default definition would be ill-formed:
  101 | class MMDEPLOY_API Registry<void> {
      |                    ^~~~~~~~~~~~~~
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/registry.h:101:20: error: use of deleted function ‘std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = mmdeploy::_registry::Registry<void>::Impl; _Dp = std::default_delete<mmdeploy::_registry::Registry<void>::Impl>]’
In file included from /usr/aarch64-linux-gnu/include/c++/10/memory:83,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/registry.h:7,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/net.h:7,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/net.cpp:3:
/usr/aarch64-linux-gnu/include/c++/10/bits/unique_ptr.h:468:7: note: declared here
  468 |       unique_ptr(const unique_ptr&) = delete;
      |       ^~~~~~~~~~
In file included from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/module.h:7,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/module.cpp:3:
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/module.cpp: In function ‘mmdeploy::_registry::Registry<mmdeploy::Module>& mmdeploy::get_registry(mmdeploy::basic_type<mmdeploy::Module>)’:
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/registry.h:211:47: error: use of deleted function ‘mmdeploy::_registry::Registry<void>::Registry(const mmdeploy::_registry::Registry<void>&)’
  211 |     static ::mmdeploy::Registry<tag> instance{};                         \
      |                                               ^
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/module.cpp:9:1: note: in expansion of macro ‘MMDEPLOY_DEFINE_REGISTRY’
    9 | MMDEPLOY_DEFINE_REGISTRY(Module);
      | ^~~~~~~~~~~~~~~~~~~~~~~~
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/registry.h:101:20: note: ‘mmdeploy::_registry::Registry<void>::Registry(const mmdeploy::_registry::Registry<void>&)’ is implicitly deleted because the default definition would be ill-formed:
  101 | class MMDEPLOY_API Registry<void> {
      |                    ^~~~~~~~~~~~~~
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/registry.h:101:20: error: use of deleted function ‘std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = mmdeploy::_registry::Registry<void>::Impl; _Dp = std::default_delete<mmdeploy::_registry::Registry<void>::Impl>]’
In file included from /usr/aarch64-linux-gnu/include/c++/10/memory:83,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/registry.h:7,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/module.h:7,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/module.cpp:3:
/usr/aarch64-linux-gnu/include/c++/10/bits/unique_ptr.h:468:7: note: declared here
  468 |       unique_ptr(const unique_ptr&) = delete;
      |       ^~~~~~~~~~
make[2]: *** [csrc/mmdeploy/core/CMakeFiles/mmdeploy_core.dir/build.make:146: csrc/mmdeploy/core/CMakeFiles/mmdeploy_core.dir/net.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [csrc/mmdeploy/core/CMakeFiles/mmdeploy_core.dir/build.make:132: csrc/mmdeploy/core/CMakeFiles/mmdeploy_core.dir/module.cpp.o] Error 1
In file included from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/model_impl.h:7,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/model.cpp:6:
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/model.cpp: In function ‘mmdeploy::_registry::Registry<mmdeploy::framework::ModelImpl>& mmdeploy::framework::get_registry(mmdeploy::basic_type<mmdeploy::framework::ModelImpl>)’:
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/registry.h:211:47: error: use of deleted function ‘mmdeploy::_registry::Registry<void>::Registry(const mmdeploy::_registry::Registry<void>&)’
  211 |     static ::mmdeploy::Registry<tag> instance{};                         \
      |                                               ^
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/model.cpp:79:1: note: in expansion of macro ‘MMDEPLOY_DEFINE_REGISTRY’
   79 | MMDEPLOY_DEFINE_REGISTRY(ModelImpl);
      | ^~~~~~~~~~~~~~~~~~~~~~~~
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/registry.h:101:20: note: ‘mmdeploy::_registry::Registry<void>::Registry(const mmdeploy::_registry::Registry<void>&)’ is implicitly deleted because the default definition would be ill-formed:
  101 | class MMDEPLOY_API Registry<void> {
      |                    ^~~~~~~~~~~~~~
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/registry.h:101:20: error: use of deleted function ‘std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = mmdeploy::_registry::Registry<void>::Impl; _Dp = std::default_delete<mmdeploy::_registry::Registry<void>::Impl>]’
In file included from /usr/aarch64-linux-gnu/include/c++/10/memory:83,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/model.h:7,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/model.cpp:3:
/usr/aarch64-linux-gnu/include/c++/10/bits/unique_ptr.h:468:7: note: declared here
  468 |       unique_ptr(const unique_ptr&) = delete;
      |       ^~~~~~~~~~
In file included from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/module.h:7,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/graph.h:7,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/graph.cpp:3:
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/graph.cpp: In function ‘mmdeploy::_registry::Registry<mmdeploy::graph::Builder>& mmdeploy::graph::get_registry(mmdeploy::basic_type<mmdeploy::graph::Builder>)’:
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/registry.h:211:47: error: use of deleted function ‘mmdeploy::_registry::Registry<void>::Registry(const mmdeploy::_registry::Registry<void>&)’
  211 |     static ::mmdeploy::Registry<tag> instance{};                         \
      |                                               ^
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/graph.cpp:130:1: note: in expansion of macro ‘MMDEPLOY_DEFINE_REGISTRY’
  130 | MMDEPLOY_DEFINE_REGISTRY(Builder);
      | ^~~~~~~~~~~~~~~~~~~~~~~~
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/registry.h:101:20: note: ‘mmdeploy::_registry::Registry<void>::Registry(const mmdeploy::_registry::Registry<void>&)’ is implicitly deleted because the default definition would be ill-formed:
  101 | class MMDEPLOY_API Registry<void> {
      |                    ^~~~~~~~~~~~~~
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/registry.h:101:20: error: use of deleted function ‘std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = mmdeploy::_registry::Registry<void>::Impl; _Dp = std::default_delete<mmdeploy::_registry::Registry<void>::Impl>]’
In file included from /usr/aarch64-linux-gnu/include/c++/10/memory:83,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/model.h:7,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/graph.h:6,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/graph.cpp:3:
/usr/aarch64-linux-gnu/include/c++/10/bits/unique_ptr.h:468:7: note: declared here
  468 |       unique_ptr(const unique_ptr&) = delete;
      |       ^~~~~~~~~~
make[2]: *** [csrc/mmdeploy/core/CMakeFiles/mmdeploy_core.dir/build.make:118: csrc/mmdeploy/core/CMakeFiles/mmdeploy_core.dir/model.cpp.o] Error 1
make[2]: *** [csrc/mmdeploy/core/CMakeFiles/mmdeploy_core.dir/build.make:216: csrc/mmdeploy/core/CMakeFiles/mmdeploy_core.dir/graph.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:905: csrc/mmdeploy/core/CMakeFiles/mmdeploy_core.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
+ make install
[  3%] Built target mmdeploy_ncnn_ops_obj
[  4%] Built target mmdeploy_ncnn_ops
[  8%] Built target spdlog
Consolidate compiler generated dependencies of target mmdeploy_core
[  9%] Building CXX object csrc/mmdeploy/core/CMakeFiles/mmdeploy_core.dir/model.cpp.o
In file included from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/model_impl.h:7,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/model.cpp:6:
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/model.cpp: In function ‘mmdeploy::_registry::Registry<mmdeploy::framework::ModelImpl>& mmdeploy::framework::get_registry(mmdeploy::basic_type<mmdeploy::framework::ModelImpl>)’:
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/registry.h:211:47: error: use of deleted function ‘mmdeploy::_registry::Registry<void>::Registry(const mmdeploy::_registry::Registry<void>&)’
  211 |     static ::mmdeploy::Registry<tag> instance{};                         \
      |                                               ^
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/model.cpp:79:1: note: in expansion of macro ‘MMDEPLOY_DEFINE_REGISTRY’
   79 | MMDEPLOY_DEFINE_REGISTRY(ModelImpl);
      | ^~~~~~~~~~~~~~~~~~~~~~~~
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/registry.h:101:20: note: ‘mmdeploy::_registry::Registry<void>::Registry(const mmdeploy::_registry::Registry<void>&)’ is implicitly deleted because the default definition would be ill-formed:
  101 | class MMDEPLOY_API Registry<void> {
      |                    ^~~~~~~~~~~~~~
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/registry.h:101:20: error: use of deleted function ‘std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = mmdeploy::_registry::Registry<void>::Impl; _Dp = std::default_delete<mmdeploy::_registry::Registry<void>::Impl>]’
In file included from /usr/aarch64-linux-gnu/include/c++/10/memory:83,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/model.h:7,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/core/model.cpp:3:
/usr/aarch64-linux-gnu/include/c++/10/bits/unique_ptr.h:468:7: note: declared here
  468 |       unique_ptr(const unique_ptr&) = delete;
      |       ^~~~~~~~~~
make[2]: *** [csrc/mmdeploy/core/CMakeFiles/mmdeploy_core.dir/build.make:118: csrc/mmdeploy/core/CMakeFiles/mmdeploy_core.dir/model.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:905: csrc/mmdeploy/core/CMakeFiles/mmdeploy_core.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
+ ls -lah install/bin/*
ls: cannot access 'install/bin/*': No such file or directory
```

**Comment by lzhangzz** [comment]
Can provide more detail on reproducing the problem? e.g. OS version, toolchain version, mmdeploy version.

We had no problem cross-compiling the SDK using clang-14 based Android toolchain or aarch64-linux-gnu-g++-9.

**Comment by Dark1Forest** [comment]
os:debain 11
toolchain:
g++-aarch64-linux-gnu is already the newest version (4:10.2.1-1).
gcc-aarch64-linux-gnu is already the newest version (4:10.2.1-1).
mmdeploy version:latest master

**Comment by lzhangzz** [comment]
Fixed in #1806, please you verify if it works for you, thanks.

**Comment by Dark1Forest** [comment]
okay, thank you

**Comment by Dark1Forest** [comment]
I modified my local code according to the code you submitted, and the error reported previously disappeared.
But now, I have another bug
`
[ 75%] Building CXX object csrc/mmdeploy/codebase/mmpose/CMakeFiles/mmdeploy_mmpose.dir/pose_tracker/track.cpp.o
In file included from /mnt/workspace/mmdeploy/csrc/mmdeploy/codebase/mmpose/pose_tracker/common.h:9,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/codebase/mmpose/pose_tracker/track.h:8,
                 from /mnt/workspace/mmdeploy/csrc/mmdeploy/codebase/mmpose/pose_tracker/track.cpp:3:
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/mpl/type_traits.h:23:19: error: ‘uint64_t’ does not name a type
   23 | using type_id_t = uint64_t;
      |                   ^~~~~~~~
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/mpl/type_traits.h:27:20: error: ‘type_id_t’ does not name a type
   27 |   static constexpr type_id_t value = 0;
      |                    ^~~~~~~~~
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/mpl/type_traits.h:32:45: error: ‘type_id_t’ does not name a type
   32 |   static constexpr auto value = static_cast<type_id_t>(-1);
      |                                             ^~~~~~~~~
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/mpl/type_traits.h:40:22: error: ‘type_id_t’ does not name a type
   40 |     static constexpr type_id_t value = id;  \
      |                      ^~~~~~~~~
/mnt/workspace/mmdeploy/csrc/mmdeploy/codebase/mmpose/pose_tracker/common.h:52:1: note: in expansion of macro ‘MMDEPLOY_REGISTER_TYPE_ID’
   52 | MMDEPLOY_REGISTER_TYPE_ID(mmdeploy_pose_tracker_param_t*, 0x32bc6919d5287035);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
/mnt/workspace/mmdeploy/csrc/mmdeploy/core/mpl/type_traits.h:40:22: error: ‘type_id_t’ does not name a type
   40 |     static constexpr type_id_t value = id;  \
      |                      ^~~~~~~~~
/mnt/workspace/mmdeploy/csrc/mmdeploy/codebase/mmpose/pose_tracker/common.h:53:1: note: in expansion of macro ‘MMDEPLOY_REGISTER_TYPE_ID’
   53 | MMDEPLOY_REGISTER_TYPE_ID(mmpose::_pose_tracker::TrackerResult, 0xacb6ddb7dc1ffbca);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [csrc/mmdeploy/codebase/mmpose/CMakeFiles/mmdeploy_mmpose.dir/build.make:188: csrc/mmdeploy/codebase/mmpose/CMakeFiles/mmdeploy_mmpose.dir/pose_tracker/track.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1489: csrc/mmdeploy/codebase/mmpose/CMakeFiles/mmdeploy_mmpose.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
`

**Comment by Dark1Forest** [comment]
I added the following code,in mmdeploy/csrc/mmdeploy/core/mpl/type_traits.h
```
#include <cstdint>
```
This library file solves the problem

**Comment by lzhangzz** [comment]
> I added the following code,in mmdeploy/csrc/mmdeploy/core/mpl/type_traits.h
> 
> ```
> #include <cstdint>
> ```
> 
> This library file solves the problem

this was fixed in #1802

**Comment by Dark1Forest** [comment]
ok ,thanks
